src/appleseed-max-impl/oslutils.cpp

```diff
@@ -30,11 +30,17 @@
 #include "oslutils.h"
 
 // appleseed-max headers.
+#include "appleseeddisneymtl/appleseeddisneymtl.h"
+#include "appleseedblendmtl/appleseedblendmtl.h"
+#include "iappleseedmtl.h"
 #include "utilities.h"
 
+
 // appleseed.renderer headers.
 #include "renderer/api/shadergroup.h"
 #include "renderer/api/utility.h"
+#include "renderer/modeling/shadergroup/shader.h"
```

Comment:
We are only allowed to include header files from `renderer/api`. You can include `renderer/api/shadergroup.h`.

Reply:
Should I put shader.h and shaderconnection.h into api\shadergroup.h?

Reply:
As far as I can tell you don't need `shader.h` nor `shaderconnection.h`.

Reply:
It iterates over shaders and connections in order to copy them to the new shader group. I can't compile it without those headers.

Reply:
Oh OK, then please do add them to the API header. Thanks